internal/runtime/process.go

```diff
@@ -254,6 +251,11 @@ func (r *runState) filterParticipants(ctx context.Context, tx *ent.Tx, limit int
 		participants = participants[:n]
 	}
 
+	if uninitialized != true {
+		l := math.Min(float64(limit), float64(len(participants)))
+		return participants[:int(l)], nil
+	}
```

Comment:
`if !unitilized` (don't need to compare a boolean) you don't know filter out uninitialized?

I think all that's needed is to have a condition that handles both opposite cases in the loop below:

```go
if (!uninitialized && participant.Uninitialized != nil && *participant.Uninitialized) || (uninitialized && (participant.Uninitialized == nil || !*participant.Uninitialized))
```